sdk/iot/src/az_iot_sas_token.c

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <az_iot_sas_token.h>
+#include <az_contract_internal.h>
+#include <az_span.h>
+
+#define LF                      '\n'
+#define AMPERSAND               '&'
+#define EQUAL_SIGN              '='
+#define SCOPE_DEVICES_STRING    "/devices/"
+#define SAS_TOKEN_SR            "SharedAccessSignature sr"
+#define SAS_TOKEN_SE            "se"
+#define SAS_TOKEN_SIG           "sig"
+#define SAS_TOKEN_SKN           "skn"
+
+az_result az_iot_sas_token_get_document(az_span iothub_fqdn, az_span device_id, int32_t expiry_time_secs, az_span* document)
```

Comment:
The document part isn't right. It needs to come in as-value and also be returned by-reference

Reply:
Updated.